src/Semantic/Api/Diffs.hs

```diff
@@ -82,84 +79,78 @@ diffGraph blobs = do
         path = T.pack $ pathForBlobPair blobPair
         lang = bridging # languageForBlobPair blobPair
 
-type DiffEffects sig m = (Member (Error SomeException) sig, Member (Reader Config) sig, Member Telemetry sig, Member Distribute sig, Member Parse sig, Carrier sig m, MonadIO m)
-
-
-dotGraphDiffParsers :: Map Language (SomeParser DOTGraphDiff Loc)
-dotGraphDiffParsers = aLaCarteParsers
-
-class DiffTerms term => DOTGraphDiff term where
-  dotGraphDiff :: (Carrier sig m, Member (Reader Config) sig) => DiffFor term Loc Loc -> m Builder
-
-instance (ConstructorName syntax, Diffable syntax, Eq1 syntax, Hashable1 syntax, Traversable syntax) => DOTGraphDiff (Term syntax) where
-  dotGraphDiff = serialize (DOT (diffStyle "diffs")) . renderTreeGraph
-
 
-jsonGraphDiffParsers :: Map Language (SomeParser JSONGraphDiff Loc)
-jsonGraphDiffParsers = aLaCarteParsers
+class DOTGraphDiff term where
+  dotGraphDiff :: (Carrier sig m, Member (Reader Config) sig, Member Telemetry sig, MonadIO m) => Edit (Blob, term Loc) (Blob, term Loc) -> m Builder
 
-class DiffTerms term => JSONGraphDiff term where
-  jsonGraphDiff :: BlobPair -> DiffFor term Loc Loc -> DiffTreeFileGraph
+instance (DiffTerms term, ConstructorName (Syntax term), Foldable (Syntax term), Functor (Syntax term)) => DOTGraphDiff term where
+  dotGraphDiff = serialize (DOT (diffStyle "diffs")) . renderTreeGraph <=< diffTerms
 
-instance (ConstructorName syntax, Diffable syntax, Eq1 syntax, Hashable1 syntax, Traversable syntax) => JSONGraphDiff (Term syntax) where
-  jsonGraphDiff blobPair diff
-    = let graph = renderTreeGraph diff
-          toEdge (Edge (a, b)) = defMessage & P.source .~ a^.diffVertexId & P.target .~ b^.diffVertexId
-          path = T.pack $ pathForBlobPair blobPair
-          lang = bridging # languageForBlobPair blobPair
-      in defMessage
-           & P.path .~ path
-           & P.language .~ lang
-           & P.vertices .~ vertexList graph
-           & P.edges .~ fmap toEdge (edgeList graph)
-           & P.errors .~ mempty
 
+class JSONGraphDiff term where
+  jsonGraphDiff :: (Carrier sig m, Member Telemetry sig, MonadIO m) => Edit (Blob, term Loc) (Blob, term Loc) -> m DiffTreeFileGraph
 
-jsonTreeDiffParsers :: Map Language (SomeParser JSONTreeDiff Loc)
-jsonTreeDiffParsers = aLaCarteParsers
-
-class DiffTerms term => JSONTreeDiff term where
-  jsonTreeDiff :: BlobPair -> DiffFor term Loc Loc -> Rendering.JSON.JSON "diffs" SomeJSON
-
-instance (Diffable syntax, Eq1 syntax, Hashable1 syntax, ToJSONFields1 syntax, Traversable syntax) => JSONTreeDiff (Term syntax) where
-  jsonTreeDiff = renderJSONDiff
+instance (DiffTerms term, ConstructorName (Syntax term), Foldable (Syntax term), Functor (Syntax term)) => JSONGraphDiff term where
```

Comment:
We still only implement the one instance, but now it’s abstracted over the term type instead of just the syntax type.